dex/src/state.rs

```diff
@@ -2686,8 +2691,201 @@ impl State {
+                order_book_state.market_state.coin_deposits_total -= native_credit;
+
+                native_taker_fee = fee_tier.taker_fee(native_credit * limit_price.get() * pc_lot_size);
+
+            }
+            Side::Ask => {
+                native_debit = coin_debit.checked_mul(coin_lot_size).unwrap();
+                native_credit = native_pc_credit;
+                deposit_vault = coin_vault.token_account();
+                payer = coin_wallet.token_account();
+                recipient = pc_wallet.token_account();
+                payer_vault = pc_vault.token_account();
+
+                order_book_state.market_state.coin_deposits_total += native_debit;
+                order_book_state.market_state.pc_deposits_total -= native_credit;
+
+                native_taker_fee = fee_tier.taker_fee(coin_debit * limit_price.get() * pc_lot_size);
+            }
+        };
+
+        let net_fees_before_referrer_rebate = native_taker_fee;
+        let referrer_rebate = fees::referrer_rebate(native_taker_fee);
+        let net_fees = net_fees_before_referrer_rebate - referrer_rebate;
+
+        order_book_state.market_state.referrer_rebates_accrued += referrer_rebate;
+        order_book_state.market_state.pc_fees_accrued += net_fees;
+        order_book_state.market_state.pc_deposits_total -= net_fees_before_referrer_rebate;
+
+        let deposit_amount = native_debit;
+        
+        if deposit_amount != 0 {
+            // desposit vault = market's PC vault (in case of bid)
+            let balance_before = deposit_vault.balance()?;
+            // transfer from payer to deposit vault
+            let deposit_instruction = spl_token::instruction::transfer(
+                &spl_token::ID,
+                payer.inner().key,
+                deposit_vault.inner().key,
+                signer.inner().key,
+                &[],
+                deposit_amount,
+            )
+            .unwrap();
+            invoke_spl_token(
+                &deposit_instruction,
+                &[
+                    payer.inner().clone(),
+                    deposit_vault.inner().clone(),
+                    signer.inner().clone(),
+                    spl_token_program.inner().clone(),
+                ],
+                &[],
+            )
+            .map_err(|err| match err {
+                ProgramError::Custom(i) => match TokenError::from_u32(i) {
+                    Some(TokenError::InsufficientFunds) => DexErrorCode::InsufficientFunds,
+                    _ => DexErrorCode::TransferFailed,
+                },
+                _ => DexErrorCode::TransferFailed,
+            })?;
+            let balance_after = deposit_vault.balance()?;
+            let balance_change = balance_after.checked_sub(balance_before);
+            check_assert_eq!(Some(deposit_amount), balance_change)?;
+        }
+
+        let nonce = order_book_state.market_state.vault_signer_nonce;
```

Comment:
Nit: I don't really see a need to declare variables that are only used once. However because `market_state` is used quite frequently I wouldn't be a opposed to declaring:
```
let market_state = &mut order_book_state.market_state;
```
near the top of this function

Reply:
(barring any mutable borrow issues)